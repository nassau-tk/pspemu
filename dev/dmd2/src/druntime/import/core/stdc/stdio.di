// D import file generated from 'src\core\stdc\stdio.d'
module core.stdc.stdio;
private 
{
    import core.stdc.config;
    import core.stdc.stddef;
    import core.stdc.stdarg;
    version (FreeBSD)
{
    import core.sys.posix.sys.types;
}
}
extern (C) nothrow 
{
    version (Windows)
{
    enum 
{
BUFSIZ = 16384,
EOF = -1,
FOPEN_MAX = 20,
FILENAME_MAX = 256,
TMP_MAX = 32767,
SYS_OPEN = 20,
}
    enum int _NFILE = 60;
    enum string _P_tmpdir = "\\";
    enum wstring _wP_tmpdir = "\\";
    enum int L_tmpnam = _P_tmpdir.length + 12;
}
else
{
    version (linux)
{
    enum 
{
BUFSIZ = 8192,
EOF = -1,
FOPEN_MAX = 16,
FILENAME_MAX = 4095,
TMP_MAX = 238328,
L_tmpnam = 20,
}
}
else
{
    version (OSX)
{
    enum 
{
BUFSIZ = 1024,
EOF = -1,
FOPEN_MAX = 20,
FILENAME_MAX = 1024,
TMP_MAX = 308915776,
L_tmpnam = 1024,
}
    private 
{
    struct __sbuf
{
    ubyte* _base;
    int _size;
}
    struct __sFILEX
{
}
}
}
else
{
    version (FreeBSD)
{
    enum 
{
EOF = -1,
FOPEN_MAX = 20,
FILENAME_MAX = 1024,
TMP_MAX = 308915776,
L_tmpnam = 1024,
}
    struct __sbuf
{
    ubyte* _base;
    int _size;
}
    alias _iobuf __sFILE;
    union __mbstate_t
{
    char[128] _mbstate8;
    long _mbstateL;
}
}
else
{
    static assert(false,"Unsupported platform");
}
}
}
}
    enum 
{
SEEK_SET,
SEEK_CUR,
SEEK_END,
}
    struct _iobuf
{
    align (1)version (Windows)
{
    char* _ptr;
    int _cnt;
    char* _base;
    int _flag;
    int _file;
    int _charbuf;
    int _bufsiz;
    int __tmpnum;
}
else
{
    version (linux)
{
    int _flags;
    char* _read_ptr;
    char* _read_end;
    char* _read_base;
    char* _write_base;
    char* _write_ptr;
    char* _write_end;
    char* _buf_base;
    char* _buf_end;
    char* _save_base;
    char* _backup_base;
    char* _save_end;
    void* _markers;
    _iobuf* _chain;
    int _fileno;
    int _blksize;
    int _old_offset;
    ushort _cur_column;
    byte _vtable_offset;
    char[1] _shortbuf;
    void* _lock;
}
else
{
    version (OSX)
{
    ubyte* _p;
    int _r;
    int _w;
    short _flags;
    short _file;
    __sbuf _bf;
    int _lbfsize;
    int* function(void*) _close;
    int* function(void*, char*, int) _read;
    fpos_t* function(void*, fpos_t, int) _seek;
    int* function(void*, char*, int) _write;
    __sbuf _ub;
    __sFILEX* _extra;
    int _ur;
    ubyte[3] _ubuf;
    ubyte[1] _nbuf;
    __sbuf _lb;
    int _blksize;
    fpos_t _offset;
}
else
{
    version (FreeBSD)
{
    ubyte* _p;
    int _r;
    int _w;
    short _flags;
    short _file;
    __sbuf _bf;
    int _lbfsize;
    void* _cookie;
    int function(void*) _close;
    int function(void*, char*, int) _read;
    fpos_t function(void*, fpos_t, int) _seek;
    int function(void*, in char*, int) _write;
    __sbuf _ub;
    ubyte* _up;
    int _ur;
    ubyte[3] _ubuf;
    ubyte[1] _nbuf;
    __sbuf _lb;
    int _blksize;
    fpos_t _offset;
    pthread_mutex_t _fl_mutex;
    pthread_t _fl_owner;
    int _fl_count;
    int _orientation;
    __mbstate_t _mbstate;
}
else
{
    static assert(false,"Unsupported platform");
}
}
}
}

}
    alias shared(_iobuf) FILE;
    enum 
{
_F_RDWR = 3,
_F_READ = 1,
_F_WRIT = 2,
_F_BUF = 4,
_F_LBUF = 8,
_F_ERR = 16,
_F_EOF = 32,
_F_BIN = 64,
_F_IN = 128,
_F_OUT = 256,
_F_TERM = 512,
}
    version (Windows)
{
    enum 
{
_IOFBF = 0,
_IOLBF = 64,
_IONBF = 4,
_IOREAD = 1,
_IOWRT = 2,
_IOMYBUF = 8,
_IOEOF = 16,
_IOERR = 32,
_IOSTRG = 64,
_IORW = 128,
_IOTRAN = 256,
_IOAPP = 512,
}
    extern shared void function() _fcloseallp;

    private extern shared FILE[_NFILE] _iob;


    shared stdin = &_iob[0];
    shared stdout = &_iob[1];
    shared stderr = &_iob[2];
    shared stdaux = &_iob[3];
    shared stdprn = &_iob[4];
}
else
{
    version (linux)
{
    enum 
{
_IOFBF = 0,
_IOLBF = 1,
_IONBF = 2,
}
    extern shared FILE* stdin;

    extern shared FILE* stdout;

    extern shared FILE* stderr;

}
else
{
    version (OSX)
{
    enum 
{
_IOFBF = 0,
_IOLBF = 1,
_IONBF = 2,
}
    private extern shared FILE* __stdinp;


    private extern shared FILE* __stdoutp;


    private extern shared FILE* __stderrp;


    alias __stdinp stdin;
    alias __stdoutp stdout;
    alias __stderrp stderr;
}
else
{
    version (FreeBSD)
{
    enum 
{
_IOFBF = 0,
_IOLBF = 1,
_IONBF = 2,
}
    private extern shared FILE* __stdinp;


    private extern shared FILE* __stdoutp;


    private extern shared FILE* __stderrp;


    alias __stdinp stdin;
    alias __stdoutp stdout;
    alias __stderrp stderr;
}
else
{
    static assert(false,"Unsupported platform");
}
}
}
}
    alias int fpos_t;
    int remove(in char* filename);
    int rename(in char* from, in char* to);
    FILE* tmpfile();
    char* tmpnam(char* s);
    int fclose(FILE* stream);
    int fflush(FILE* stream);
    FILE* fopen(in char* filename, in char* mode);
    FILE* freopen(in char* filename, in char* mode, FILE* stream);
    void setbuf(FILE* stream, char* buf);
    int setvbuf(FILE* stream, char* buf, int mode, size_t size);
    int fprintf(FILE* stream, in char* format,...);
    int fscanf(FILE* stream, in char* format,...);
    int sprintf(char* s, in char* format,...);
    int sscanf(in char* s, in char* format,...);
    int vfprintf(FILE* stream, in char* format, va_list arg);
    int vfscanf(FILE* stream, in char* format, va_list arg);
    int vsprintf(char* s, in char* format, va_list arg);
    int vsscanf(in char* s, in char* format, va_list arg);
    int vprintf(in char* format, va_list arg);
    int vscanf(in char* format, va_list arg);
    int printf(in char* format,...);
    int scanf(in char* format,...);
    int fgetc(FILE* stream);
    int fputc(int c, FILE* stream);
    char* fgets(char* s, int n, FILE* stream);
    int fputs(in char* s, FILE* stream);
    char* gets(char* s);
    int puts(in char* s);
    extern (D) 
{
    int getchar()
{
return getc(stdin);
}
    int putchar(int c)
{
return putc(c,stdout);
}
    int getc(FILE* stream)
{
return fgetc(stream);
}
    int putc(int c, FILE* stream)
{
return fputc(c,stream);
}
}
    int ungetc(int c, FILE* stream);
    size_t fread(void* ptr, size_t size, size_t nmemb, FILE* stream);
    size_t fwrite(in void* ptr, size_t size, size_t nmemb, FILE* stream);
    int fgetpos(FILE* stream, fpos_t* pos);
    int fsetpos(FILE* stream, in fpos_t* pos);
    int fseek(FILE* stream, c_long offset, int whence);
    c_long ftell(FILE* stream);
    version (Windows)
{
    extern (D) 
{
    void rewind(FILE* stream)
{
fseek(stream,0L,SEEK_SET);
stream._flag &= ~_IOERR;
}
    void clearerr(FILE* stream)
{
stream._flag &= ~(_IOERR | _IOEOF);
}
    int feof(FILE* stream)
{
return stream._flag & _IOEOF;
}
    int ferror(FILE* stream)
{
return stream._flag & _IOERR;
}
}
    int _snprintf(char* s, size_t n, in char* fmt,...);
    alias _snprintf snprintf;
    int _vsnprintf(char* s, size_t n, in char* format, va_list arg);
    alias _vsnprintf vsnprintf;
}
else
{
    version (linux)
{
    void rewind(FILE* stream);
    void clearerr(FILE* stream);
    int feof(FILE* stream);
    int ferror(FILE* stream);
    int fileno(FILE*);
    int snprintf(char* s, size_t n, in char* format,...);
    int vsnprintf(char* s, size_t n, in char* format, va_list arg);
}
else
{
    version (OSX)
{
    void rewind(FILE*);
    void clearerr(FILE*);
    int feof(FILE*);
    int ferror(FILE*);
    int fileno(FILE*);
    int snprintf(char* s, size_t n, in char* format,...);
    int vsnprintf(char* s, size_t n, in char* format, va_list arg);
}
else
{
    version (FreeBSD)
{
    void rewind(FILE*);
    void clearerr(FILE*);
    int feof(FILE*);
    int ferror(FILE*);
    int fileno(FILE*);
    int snprintf(char* s, size_t n, in char* format,...);
    int vsnprintf(char* s, size_t n, in char* format, va_list arg);
}
else
{
    static assert(false,"Unsupported platform");
}
}
}
}
    void perror(in char* s);
}

