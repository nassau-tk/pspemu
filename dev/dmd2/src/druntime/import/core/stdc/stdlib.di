// D import file generated from 'src\core\stdc\stdlib.d'
module core.stdc.stdlib;
private import core.stdc.config;

public import core.stdc.stddef;

extern (C) nothrow 
{
    struct div_t
{
    int quot;
    int rem;
}
    struct ldiv_t
{
    int quot;
    int rem;
}
    struct lldiv_t
{
    long quot;
    long rem;
}
    enum EXIT_SUCCESS = 0;
    enum EXIT_FAILURE = 1;
    enum MB_CUR_MAX = 1;
    version (Windows)
{
    enum RAND_MAX = 32767;
}
else
{
    version (linux)
{
    enum RAND_MAX = 2147483647;
}
else
{
    version (OSX)
{
    enum RAND_MAX = 2147483647;
}
else
{
    version (FreeBSD)
{
    enum RAND_MAX = 2147483647;
}
else
{
    version (Solaris)
{
    enum RAND_MAX = 32767;
}
else
{
    static assert(false,"Unsupported platform");
}
}
}
}
}
    double atof(in char* nptr);
    int atoi(in char* nptr);
    c_long atol(in char* nptr);
    long atoll(in char* nptr);
    double strtod(in char* nptr, char** endptr);
    float strtof(in char* nptr, char** endptr);
    real strtold(in char* nptr, char** endptr);
    c_long strtol(in char* nptr, char** endptr, int base);
    long strtoll(in char* nptr, char** endptr, int base);
    c_ulong strtoul(in char* nptr, char** endptr, int base);
    ulong strtoull(in char* nptr, char** endptr, int base);
    int rand();
    void srand(uint seed);
    void* malloc(size_t size);
    void* calloc(size_t nmemb, size_t size);
    void* realloc(void* ptr, size_t size);
    void free(void* ptr);
    void abort();
    void exit(int status);
    int atexit(void function() func);
    void _Exit(int status);
    char* getenv(in char* name);
    int system(in char* string);
    void* bsearch(in void* key, in void* base, size_t nmemb, size_t size, int function(in void*, in void*) compar);
    void qsort(void* base, size_t nmemb, size_t size, int function(in void*, in void*) compar);
    pure int abs(int j);

    pure c_long labs(c_long j);

    pure long llabs(long j);

    div_t div(int numer, int denom);
    ldiv_t ldiv(c_long numer, c_long denom);
    lldiv_t lldiv(long numer, long denom);
    int mblen(in char* s, size_t n);
    int mbtowc(wchar_t* pwc, in char* s, size_t n);
    int wctomb(char* s, wchar_t wc);
    size_t mbstowcs(wchar_t* pwcs, in char* s, size_t n);
    size_t wcstombs(char* s, in wchar_t* pwcs, size_t n);
    version (DigitalMars)
{
    void* alloca(size_t size);
}
}

