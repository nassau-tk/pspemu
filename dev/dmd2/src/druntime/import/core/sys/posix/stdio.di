// D import file generated from 'src\core\sys\posix\stdio.d'
module core.sys.posix.stdio;
private import core.sys.posix.config;

public import core.stdc.stdio;

public import core.sys.posix.sys.types;

extern (C) 
{
    version (linux)
{
    static if(__USE_LARGEFILE64)
{
    int fgetpos64(FILE*, fpos_t*);
    alias fgetpos64 fgetpos;
    FILE* fopen64(in char*, in char*);
    alias fopen64 fopen;
    FILE* freopen64(in char*, in char*, FILE*);
    alias freopen64 freopen;
    int fseek64(FILE*, c_long, int);
    alias fseek64 fseek;
    int fsetpos64(FILE*, in fpos_t*);
    alias fsetpos64 fsetpos;
    FILE* tmpfile64();
    alias tmpfile64 tmpfile;
}
else
{
    int fgetpos(FILE*, fpos_t*);
    FILE* fopen(in char*, in char*);
    FILE* freopen(in char*, in char*, FILE*);
    int fseek(FILE*, c_long, int);
    int fsetpos(FILE*, in fpos_t*);
    FILE* tmpfile();
}
}
    version (linux)
{
    enum L_ctermid = 9;
    static if(__USE_FILE_OFFSET64)
{
    int fseeko64(FILE*, off_t, int);
    alias fseeko64 fseeko;
}
else
{
    int fseeko(FILE*, off_t, int);
}
    static if(__USE_LARGEFILE64)
{
    off_t ftello64(FILE*);
    alias ftello64 ftello;
}
else
{
    off_t ftello(FILE*);
}
}
else
{
    version (Posix)
{
    int fseeko(FILE*, off_t, int);
    off_t ftello(FILE*);
}
}
    version (Posix)
{
    char* ctermid(char*);
    FILE* fdopen(int, in char*);
    int fileno(FILE*);
    char* gets(char*);
    int pclose(FILE*);
    FILE* popen(in char*, in char*);
}
    version (linux)
{
    void flockfile(FILE*);
    int ftrylockfile(FILE*);
    void funlockfile(FILE*);
    int getc_unlocked(FILE*);
    int getchar_unlocked();
    int putc_unlocked(int, FILE*);
    int putchar_unlocked(int);
}
    version (linux)
{
    enum P_tmpdir = "/tmp";
    char* tempnam(in char*, in char*);
}
}
