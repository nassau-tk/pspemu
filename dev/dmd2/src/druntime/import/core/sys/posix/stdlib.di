// D import file generated from 'src\core\sys\posix\stdlib.d'
module core.sys.posix.stdlib;
private import core.sys.posix.config;

public import core.stdc.stdlib;

public import core.sys.posix.sys.wait;

extern (C) 
{
    version (linux)
{
    int posix_memalign(void**, size_t, size_t);
}
else
{
    version (FreeBSD)
{
    int posix_memalign(void**, size_t, size_t);
}
}
    version (linux)
{
    int setenv(in char*, in char*, int);
    int unsetenv(in char*);
    void* valloc(size_t);
}
else
{
    version (OSX)
{
    int setenv(in char*, in char*, int);
    int unsetenv(in char*);
    void* valloc(size_t);
}
else
{
    version (FreeBSD)
{
    int setenv(in char*, in char*, int);
    int unsetenv(in char*);
    void* valloc(size_t);
}
}
}
    version (linux)
{
    int rand_r(uint*);
}
else
{
    version (OSX)
{
    int rand_r(uint*);
}
else
{
    version (FreeBSD)
{
    int rand_r(uint*);
}
}
}
    version (linux)
{
    c_long a64l(in char*);
    double drand48();
    char* ecvt(double, int, int*, int*);
    double erand48(ref ushort[3]);
    char* fcvt(double, int, int*, int*);
    char* gcvt(double, int, char*);
    int getsubopt(char**, in char**, char**);
    int grantpt(int);
    char* initstate(uint, char*, size_t);
    c_long jrand48(ref ushort[3]);
    char* l64a(c_long);
    void lcong48(ref ushort[7]);
    c_long lrand48();
    char* mktemp(char*);
    c_long mrand48();
    c_long nrand48(ref ushort[3]);
    int posix_openpt(int);
    char* ptsname(int);
    int putenv(char*);
    c_long random();
    char* realpath(in char*, char*);
    ushort seed48(ref ushort[3]);
    void setkey(in char*);
    char* setstate(in char*);
    void srand48(c_long);
    void srandom(uint);
    int unlockpt(int);
    static if(__USE_LARGEFILE64)
{
    int mkstemp64(char*);
    alias mkstemp64 mkstemp;
}
else
{
    int mkstemp(char*);
}
}
else
{
    version (OSX)
{
    c_long a64l(in char*);
    double drand48();
    char* ecvt(double, int, int*, int*);
    double erand48(ref ushort[3]);
    char* fcvt(double, int, int*, int*);
    char* gcvt(double, int, char*);
    int getsubopt(char**, in char**, char**);
    int grantpt(int);
    char* initstate(uint, char*, size_t);
    c_long jrand48(ref ushort[3]);
    char* l64a(c_long);
    void lcong48(ref ushort[7]);
    c_long lrand48();
    char* mktemp(char*);
    int mkstemp(char*);
    c_long mrand48();
    c_long nrand48(ref ushort[3]);
    int posix_openpt(int);
    char* ptsname(int);
    int putenv(char*);
    c_long random();
    char* realpath(in char*, char*);
    ushort seed48(ref ushort[3]);
    void setkey(in char*);
    char* setstate(in char*);
    void srand48(c_long);
    void srandom(uint);
    int unlockpt(int);
}
else
{
    version (FreeBSD)
{
    c_long a64l(in char*);
    double drand48();
    double erand48(ref ushort[3]);
    int getsubopt(char**, in char**, char**);
    int grantpt(int);
    char* initstate(uint, char*, size_t);
    c_long jrand48(ref ushort[3]);
    char* l64a(c_long);
    void lcong48(ref ushort[7]);
    c_long lrand48();
    char* mktemp(char*);
    int mkstemp(char*);
    c_long mrand48();
    c_long nrand48(ref ushort[3]);
    int posix_openpt(int);
    char* ptsname(int);
    int putenv(char*);
    c_long random();
    char* realpath(in char*, char*);
    ushort seed48(ref ushort[3]);
    void setkey(in char*);
    char* setstate(in char*);
    void srand48(c_long);
    void srandom(uint);
    int unlockpt(int);
}
}
}
}
