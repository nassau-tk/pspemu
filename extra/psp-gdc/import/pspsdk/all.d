module pspsdk.all;

public import
	pspsdk.pspkernel,
	pspsdk.pspkerneltypes,

	// controller
	pspsdk.pspctrl,

	// graphics
	pspsdk.pspdisplay,
	pspsdk.pspge, pspsdk.pspgu, pspsdk.pspgum,
	
	// audio
	pspsdk.pspaudio,
	pspsdk.pspaudiolib,

	// utils
	pspsdk.pspthreadman,
	pspsdk.psploadexec,
	pspsdk.pspdebug,
	pspsdk.psputils,
	
	// d utilities
	pspsdk.utils.callback,
	pspsdk.utils.emits,
	pspsdk.utils.vram
;
