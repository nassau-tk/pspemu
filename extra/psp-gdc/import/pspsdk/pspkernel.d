module pspsdk.pspkernel;

/*
#include <pspuser.h>
#include <pspiofilemgr_kernel.h>
#include <psploadcore.h>
#include <pspstdio_kernel.h>
#include <pspsysreg.h>
#include <pspkdebug.h>
#include <pspintrman_kernel.h>
#include <pspmodulemgr_kernel.h>
*/

/**
 * Set the $pc register to a kernel memory address.
 *
 * When the PSP's kernel library stubs are called, they expect to be accessed
 * from the kernel's address space.  Use this function to set $pc to the kernel
 * address space, before calling a kernel library stub.
 */
/*
#define pspKernelSetKernelPC()  \
{     \
	__asm__ volatile (      \
	"la     $8, 1f\n\t"     \
	"lui    $9, 0x8000\n\t" \
	"or     $8, $9\n\t"     \
	"jr     $8\n\t"         \
	" nop\n\t"              \
	"1:\n\t"                \
	: : : "$8", "$9");      \
	sceKernelIcacheClearAll(); \
}
*/

/**
 * Set the $pc register to a user memory address.
 *
 */
/*
#define pspKernelSetUserPC()  \
{     \
	__asm__ volatile (      \
	"la     $8, 1f\n\t"     \
	"li     $9, 0x7FFFFFFF\n\t" \
	"and    $8, $9\n\t"     \
	"jr     $8\n\t"         \
	" nop\n\t"              \
	"1:\n\t"                \
	: : : "$8", "$9");      \
	sceKernelIcacheClearAll(); \
}
*/