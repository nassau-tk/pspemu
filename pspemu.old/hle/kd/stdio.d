module pspemu.hle.kd.stdio; // kd/stdio.prx (sceStdio)

import pspemu.hle.Module;

class StdioForUser : Module {
	void initNids() {
		mixin(registerd!(0x172D316E, sceKernelStdin));
		mixin(registerd!(0xA6BAB2E9, sceKernelStdout));
		mixin(registerd!(0xF78BA90A, sceKernelStderr));
		mixin(registerd!(0x98220F3E, sceKernelStdoutReopen));
		mixin(registerd!(0xFB5380C5, sceKernelStderrReopen));
	}

	/**
	 * Function to get the current standard in file no
	 * 
	 * @return The stdin fileno
	 */
	SceUID sceKernelStdin() { return STDIN; }

	/**
	 * Function to get the current standard out file no
	 * 
	 * @return The stdout fileno
	 */
	SceUID sceKernelStdout() { return STDOUT; }

	/**
	 * Function to get the current standard err file no
	 * 
	 * @return The stderr fileno
	 */
	SceUID sceKernelStderr() { return STDERR; }

	/** 
	 * Function reopen the stdout file handle to a new file
	 *
	 * @param file - The file to open.
	 * @param flags - The open flags 
	 * @param mode - The file mode
	 * 
	 * @return < 0 on error.
	 */
	int sceKernelStdoutReopen(string file, int flags, SceMode mode) {
		unimplemented();
		return -1;
	}

	/** 
	 * Function reopen the stderr file handle to a new file
	 *
	 * @param file - The file to open.
	 * @param flags - The open flags 
	 * @param mode - The file mode
	 * 
	 * @return < 0 on error.
	 */
	int sceKernelStderrReopen(string file, int flags, SceMode mode) {
		unimplemented();
		return -1;
	}
}

class StdioForKernel : StdioForUser {
}

enum : SceUID {
	STDIN  = -1,
	STDOUT = -2,
	STDERR = -3
}

static this() {
	mixin(Module.registerModule("StdioForKernel"));
	mixin(Module.registerModule("StdioForUser"));
}