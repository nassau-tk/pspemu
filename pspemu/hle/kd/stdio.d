module pspemu.hle.kd.stdio; // kd/stdio.prx (sceStdio)

import pspemu.hle.Module;

class StdioForKernel : Module {
}

class StdioForUser : StdioForKernel {
}

static this() {
	mixin(Module.registerModule("StdioForKernel"));
	mixin(Module.registerModule("StdioForUser"));
}