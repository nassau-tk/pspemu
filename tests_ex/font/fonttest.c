#include <pspsdk.h>
#include <pspkernel.h>
#include <stdio.h>
#include <stdlib.h>
#include "libfont.h"

PSP_MODULE_INFO("font test", 0, 1, 1);
PSP_MAIN_THREAD_ATTR(THREAD_ATTR_USER);

#define eprintf(...) pspDebugScreenPrintf(__VA_ARGS__); Kprintf(__VA_ARGS__);

static void *Font_Alloc(void *data, u32 size) {
	eprintf("Font_Alloc(%08X, %d)\n", (uint)data, size);
	return malloc(size);
}

static void Font_Free(void *data, void *p){
	eprintf("Font_Free(%08X, %08X)\n", (uint)data, (uint)p);
	free(p);
}

int main(int argc, char *argv[]) {
	FontLibraryHandle libHandle;
	FontHandle        fontHandle;
	FontInfo          fontInfo;
	int result;
	u32 errorCode;
	FontNewLibParams params = { NULL, 4, NULL, Font_Alloc, Font_Free, NULL, NULL, NULL, NULL, NULL, NULL };

	pspDebugScreenInit();
	
	eprintf("Starting fonttest...\n");

	libHandle = sceFontNewLib(&params, &errorCode);
	eprintf("sceFontNewLib: %d, %08X\n", libHandle != 0, errorCode);
	
	fontHandle = sceFontOpen(libHandle, 0, 0777, &errorCode);
	eprintf("sceFontOpen: %d, %08X\n", fontHandle != 0, errorCode);
	
	/*
	result = sceFontGetFontInfo(fontHandle, &fontInfo);
	eprintf("sceFontGetFontInfo: %d\n", result);
	
	#define PRINT_GLYPH_METRICS_MAX(TYPE) eprintf("fontInfo.max" #TYPE ": %d, %f\n", fontInfo.max## TYPE ## I, fontInfo.max## TYPE ## F);
	#define PRINT_GLYPH_METRICS_MIN(TYPE) eprintf("fontInfo.min" #TYPE ": %d, %f\n", fontInfo.min## TYPE ## I, fontInfo.min## TYPE ## F);
	
	PRINT_GLYPH_METRICS_MAX(GlyphWidth);
	PRINT_GLYPH_METRICS_MAX(GlyphHeight);
	PRINT_GLYPH_METRICS_MAX(GlyphAscender);
	PRINT_GLYPH_METRICS_MAX(GlyphDescender);
	PRINT_GLYPH_METRICS_MAX(GlyphLeftX);
	PRINT_GLYPH_METRICS_MAX(GlyphBaseY);
	PRINT_GLYPH_METRICS_MIN(GlyphCenterX);
	PRINT_GLYPH_METRICS_MAX(GlyphTopY);
	PRINT_GLYPH_METRICS_MAX(GlyphAdvanceX);
	PRINT_GLYPH_METRICS_MAX(GlyphAdvanceY);
	*/

	return 0;
}