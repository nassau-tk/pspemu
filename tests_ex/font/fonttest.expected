Starting fonttest...
sceFontNewLib: 1, 00000000
sceFontOpen: 1, 00000000